presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -80,6 +80,15 @@ public SourcePartitionedScheduler(
         this.partitionedNode = partitionedNode;
     }
 
+    /**
+     * Schedules as much work as possible without blocking.
+     * <p>
+     * Before setting the finished flag, at least one split is guaranteed to be scheduled.
+     * In the event that no ordinary split is available from the underlying SplitSource,
+     * a synthesized EmptySplit will be scheduled.
+     * This at-least-one-split guarantee is provided on a per-SplitSource basis,
+     * not per-DriverGroup basis.
```

Comment:
I don't think there's a "DriverGroup" concept at this point, so defer this part of the javadoc until it's introduced.

Reply:
I'll move this commit after "Support addressable split group in SplitSource/Manager"